packages/app/src/components/SourceForm.tsx

```diff
@@ -1032,7 +974,7 @@ export function TableSourceForm({
 
       createSource.mutate(
         // TODO: HDX-1768 get rid of this type assertion
-        { source: data as TSource },
+        { source: parseResult.data as TSource },
```

Comment:
Calling out that this change results in us only saving the specific fields which are expected for a particular source kind (since we are saving the Zod-parsed object, instead of the raw data object, which usually contains other fields from other source kinds and from the schema inference function). If we are intentionally saving the raw data object, happy to change this back!

Reply:
Let's revert this piece and create another ticket (tech debt). I'd like to isolate your actual changes and remove the possibility of unrelated regressions.